### DISCUSSION TITLE
How to bind to Unix domain socket instead of localhost

### DISCUSSION THREAD

**Discussion by flux77** [Kozea/Radicale#1362] [Q&A general]
Is it possible to bind Radicale to a Unix domain socket instead of localhost? I tried this in my configuration file:
```
[server]
hosts = /tmp/radicale.sock
```

But Radicale refuses to start. It gives this error message:

```
Invalid configuration: Invalid list_of_ip_address value for option 'hosts' in section 'server' in config file '/usr/local/etc/radicale/config': '/tmp/radicale.sock'
```

Is there a correct way to do this?

**Comment by pbiering** [comment]
"radicale" is a network service, what would be the purpose for using a "unix domain socket"?

**Comment by Unrud** [comment]
The built-in WSGI server is minimalistic and doesn't support it. You have to use something like uWSGI or Gunicorn.

@pbiering For example, it can be useful in combination with a local reverse proxy. You don't need to expose a port on the loopback interface.

**Comment by pbiering** [comment]
Again what learnt, found also that Apache is supporting it:

https://stackoverflow.com/questions/51243332/apache-proxy-pass-to-unix-domain-socket

Can be helpful to prevent at least sniffing on loopback interface e.g. with `ngrep` for potentially unencrypted content. But potentially the SELinux policy must be adjusted beside adding support in `radicale`

**Comment by emilfihlman** [comment]
I second the need for unix domain socket support. Adding support for it is easy in Python and it's more secure and flexible.

**Comment by pbiering** [comment]
According to @Unrud it is not possible at the moment -> one has to investigate how to support and contribute

**Comment by flux77** [comment]
Why was this issue closed? The issue is still present in Radicale.
@pbiering

**Comment by pbiering** [comment]
@flux77 : discussion reopend but requires contribution.
